doc/tutorials/platformer/app/lib/ember_quest.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:ember_quest/actors/ember.dart';
+import 'package:ember_quest/actors/water_enemy.dart';
+import 'package:ember_quest/managers/segment_manager.dart';
+import 'package:ember_quest/objects/ground_block.dart';
+import 'package:ember_quest/objects/platform_block.dart';
+import 'package:ember_quest/objects/star.dart';
+import 'package:ember_quest/overlays/hud.dart';
+
+import 'package:flame/events.dart';
+import 'package:flame/game.dart';
+import 'package:flutter/material.dart';
+
+class EmberQuestGame extends FlameGame
+    with HasCollisionDetection, HasKeyboardHandlerComponents {
+  EmberQuestGame();
+
+  late EmberPlayer _ember;
+  late double lastBlockXPosition = 0.0;
+  late GlobalKey lastBlockKey;
```

Comment:
What is `GlobalKey` used for?

Reply:
It's used to give just the last block in each segment a unique key for comparison. I tried using the classes key but was having inconsistent results.  That was before I changed the constructor as you suggested so it may not be necessary. I will need to test again.